src/daemon/rpm.c

```diff
@@ -75,19 +75,9 @@ void rpm_destroy()
     rpmFreeCrypto();
     rpmFreeMacros(NULL);
     rpmFreeRpmrc();
-
-/* rpm >= 4.14 handles this automatically on exit */
-#if 0
-    /* RPM doc says "clean up any open iterators and databases".
-     * Observed to eliminate these Berkeley DB warnings:
-     * "BDB2053 Freeing read locks for locker 0x1e0: 28718/139661746636736"
-     */
-    rpmdbCheckTerminate(1);
-#endif
 #endif
 
-    libreport_list_free_with_free(list_fingerprints);
-    list_fingerprints = NULL;
+    g_list_free_full(g_steal_pointer(&list_fingerprints), free);
```

Comment:
Would `g_clear_list()` not be better in this case?

Reply:
Hmm, not for Fedora 31, at least.

Reply:
There's always this one little function that I hadn't heard about which does that one more step.

Well, we could wait two months and “improve” the code even further once F31 is gone. I'm okay wit that for now.

Reply:
Not to brag (much), but it’s a rather new thing: https://gitlab.gnome.org/GNOME/glib/commit/58ba7d78fbd7ecb4c0df2dc7e251627ebbffb9d5

We can wait, I have no preference.